src/api_2.rs

```diff
@@ -0,0 +1,207 @@
+extern crate rocksdb;
+
+use std::{
+	pin::Pin,
+	sync::Arc,
+	task::{Context, Poll},
+};
+
+use ::futures::prelude::*;
+use chrono::{DateTime, Local};
+use hyper::{
+	header::ACCESS_CONTROL_ALLOW_ORIGIN, service::Service, Body, Method, Request, Response, Server,
+	StatusCode,
+};
+use regex::Regex;
+use rocksdb::{DBWithThreadMode, SingleThreaded};
+use tokio;
+
+use crate::{
+	rpc::{
+		check_http, generate_app_specific_cells, get_block_by_number, get_kate_query_proof_by_cell,
+	},
+	types::{Block, Cell, Header},
+};
+
+// 💡 HTTP part where handles the RPC Queries
+
+//service part of hyper
+struct Handler {
+	store: Arc<DBWithThreadMode<SingleThreaded>>,
+	url: String,
+}
+
+impl Service<Request<Body>> for Handler {
+	type Error = hyper::Error;
+	type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send>>;
+	type Response = Response<Body>;
+
+	fn poll_ready(&mut self, _: &mut Context) -> Poll<Result<(), Self::Error>> {
+		Poll::Ready(Ok(()))
+	}
+
+	fn call(&mut self, req: Request<Body>) -> Self::Future {
+		fn match_url_block(path: &str) -> Result<u64, String> {
```

Comment:
Should we extract these functions and have unit tests for them externally? 

Do we need unwraps here or some more appropriate error handling?

Reply:
well fn call() is needed which used by the hyper [crate](https://docs.rs/hyper/latest/hyper/service/trait.Service.html). Atleast thats my understanding

Reply:
@kroos47 ofc, but talking about the `match_url_block` func...

Reply:
oh okey. we can do that 👍